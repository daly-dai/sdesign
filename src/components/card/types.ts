import { CardProps } from 'antd';
import { ReactNode } from 'react';

export interface SCardProps extends Omit<CardProps, 'children'> {
  children?: ReactNode;
  hasBottomPadding?: boolean; // 底部是否包含边距
}
