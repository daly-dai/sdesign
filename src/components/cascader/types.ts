import { CascaderProps } from 'antd';

export type SingleValueType = (string | number)[];
export type ValueType = SingleValueType | SingleValueType[] | undefined;

export type SCascaderProps = Omit<
  CascaderProps<any>,
  'onChange' | 'value' | 'defaultValue' | 'multiple'
> & {
  onChange?: (value: string, selectOptions?: any[]) => void;
  value?: string | null | ValueType;
  options?: any[];
  defaultValue?: string | ValueType;
  multiple?: boolean;
};
