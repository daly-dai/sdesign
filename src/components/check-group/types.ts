import { Checkbox } from 'antd';
import { ComponentProps, HTMLAttributes, ReactNode } from 'react';

import { ExtraComType } from '@dalydb/sdesign/types/base';

export type CheckboxGroupType = HTMLAttributes<object> &
  Omit<ComponentProps<typeof Checkbox.Group>, 'onChange' | 'value'>;
export type CheckboxValueType = any;

export type SCheckGroupProps = CheckboxGroupType &
  ExtraComType & {
    value?: CheckboxValueType[] | undefined | string;
    onChange?: (value: string | undefined) => void;
    children?: ReactNode;
  };
