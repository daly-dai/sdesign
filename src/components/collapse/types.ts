import { ButtonProps } from 'antd/es/button/button';

export interface SCollapseProps extends ButtonProps {
  collapse?: boolean;
  setCollapse?: (collapse: boolean) => void;
  onExpand?: (collapse: boolean) => void;
}
