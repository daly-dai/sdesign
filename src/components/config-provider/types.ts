import { ReactNode } from 'react';

export type SConfigProviderType = {
  // 全部的字典
  globalDict?: Record<string, any>;
  // 上传组件接口地址地址
  uploadUrl?: string;
  children?: ReactNode;
  prefixCls?: string;
};

export interface ConfigContextProps {
  globalDict?: Record<string, Record<string, any> | any>;
  uploadUrl?: string;
  getPrefixCls: (suffixCls?: string, customizePrefixCls?: string) => string;
}
