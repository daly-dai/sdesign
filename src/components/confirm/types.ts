import { ButtonProps, PopconfirmProps } from 'antd';

import { tuple } from '@dalydb/sdesign/utils/types';

const ConfirmTypes = tuple('pop', 'modal');
export type ConfirmType = (typeof ConfirmTypes)[number];

export interface SConfirmProps {
  onConfirm?: () => void; // 确认的回调函数
  onCancel?: () => void; // 取消的回调函数
  type?: ConfirmType;
  text?: string;
  title?: React.ReactNode;
  description?: React.ReactNode;
  children?: React.ReactNode;
  popConfirmProps?: PopconfirmProps; // 弹出框的属性传参
  buttonProps?: ButtonProps; // 按钮的属性传参
  disabled?: boolean;
}
