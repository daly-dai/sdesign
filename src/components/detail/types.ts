import { DescriptionsProps } from 'antd';
import { DescriptionsItemType } from 'antd/es/descriptions';
import React, { ReactNode } from 'react';

import { FileListProps, PageTitleProps } from '@daly/sdesign';
import { tuple } from '@daly/sdesign/utils/types';

const ItemTypes = tuple(
  'text',
  'empty',
  'file',
  'dict',
  'img',
  'rangeTime',
  'checkbox',
  'placeholder',
);

export type ItemType = (typeof ItemTypes)[number];

export interface DictReflect {
  label?: string;
  name?: string;
}

export type SDetailItemType = DetailItemType & {
  label?: ReactNode;
  name?: string | string[];
  render?: (value?: any, dataSource?: any) => ReactNode;
  span?: number;
  /** * @desc 隐藏该属性 */
  hidden?: boolean;
};

export type DetailItemType = {
  type?: ItemType;
  fileProps?: Partial<FileListProps>;
  dictReflect?: DictReflect;
  dictKey?: string;
  value?: any;
  dictMap?: Record<string, string> | any[] | null;
  render?: (value?: any, dataSource?: any) => ReactNode;
  dataSource?: Record<string, any>;
};

export type SDetailItem = SDetailItemType &
  Omit<DescriptionsItemType, 'children' | 'key'>;

export interface SDetailProps
  extends Omit<DescriptionsProps, 'items' | 'title'> {
  titleDesc?: ReactNode;
  titleAction?: ReactNode;
  dataSource?: Record<string, any>;
  items?: SDetailItem[];
  labelStyle?: React.CSSProperties;
  contentStyle?: React.CSSProperties;
  hasCardBg?: boolean;
  container?: React.ComponentType<any>;
  title?: string | ReactNode;
  /** @description dataSource中的key */
  detailName?: string;
}

export interface SDetailGroupItem {
  groupTitle?: string | ReactNode;
  groupTitleProps?: Omit<PageTitleProps, 'title'>;
  groupContainer?: React.ComponentType<any>;
  groupItems?: SDetailProps[];
  items?: SDetailItem[];
  itemProps?: Omit<SDetailProps, 'items' | 'dataSource'>;
  dataSource?: Record<string, any>;
  /*** @desc 隐藏该属性*/
  hidden?: boolean;
}

export interface SDetailGroupProps {
  items?: SDetailGroupItem[];
  dataSource?: Record<string, any>;
}
