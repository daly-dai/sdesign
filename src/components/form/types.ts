import {
  Cascader,
  Checkbox,
  ColProps,
  DatePicker,
  FormItemProps,
  FormProps,
  Input,
  InputNumber,
  Radio,
  RowProps,
  Slider,
  Switch,
  Table,
  TimePicker,
  TreeSelect,
  Upload,
} from 'antd';
import { NamePath } from 'antd/es/form/interface';
import React, { ComponentProps, HTMLAttributes, ReactNode } from 'react';

import SCascader from '../cascader';
import SCheckGroup from '../check-group';
import SDatePicker from '../date-picker';
import SDatePickerRange from '../date-picker-range';
import SDependency from '../dependency';
import { RenderChildren } from '../dependency/types';
import SRadioGroup from '../radio-group';
import SSelect from '../select';
import SUpload from '../upload';
import InternalUpload from '../upload/instance';

import { RegKeyType } from '@dalydb/sdesign/types/reg';

export type FormFieldMapType = {
  input: typeof Input;
  inputNumber: typeof InputNumber;
  password: typeof Input.Password;
  textarea: typeof Input.TextArea;
  select: typeof SSelect;
  slider: typeof Slider;
  radio: typeof Radio;
  radioGroup: typeof SRadioGroup;
  switch: typeof Switch;
  treeSelect: typeof TreeSelect;
  upload: typeof Upload;
  SUpload: typeof InternalUpload;
  SUploadDragger: typeof SUpload.Dragger;
  SUploadPicture: typeof SUpload.Picture;
  datePicker: typeof DatePicker;
  SDatePicker: typeof SDatePicker;
  datePickerRange: typeof DatePicker.RangePicker;
  SDatePickerRange: typeof SDatePickerRange;
  timePicker: typeof TimePicker;
  timePickerRange: typeof TimePicker.RangePicker;
  checkbox: typeof Checkbox;
  checkGroup: typeof SCheckGroup;
  cascader: typeof Cascader;
  SCascader: typeof SCascader;
  table: typeof Table;
  dependency: typeof SDependency;
};

export type FormComType = keyof FormFieldMapType;

export type FormComPropsType = HTMLAttributes<object> &
  ComponentProps<FormFieldMapType[FormComType]>;

export type FormItemType = FormComType | 'placeholder';
export interface ItemsProps
  extends Omit<FormItemProps, 'label | name' | 'required'> {
  label?: ReactNode;
  name?: NamePath;
  style?: React.CSSProperties;
  // 组件的类型
  type?: FormItemType;
  // 依赖的字段，只在type为dependency时生效
  depNames?: string[];
  // 组件的配置项
  fieldProps?: FormComPropsType;
  // 自定义组件
  customCom?: ReactNode | RenderChildren<any>;
  // 校验规则
  regKey?: RegKeyType;
  // 自定义渲染
  render?: RenderChildren<any>;
  // 是否为必填
  required?: string | boolean;
  // 只读模式
  readonly?: boolean;
  // 嵌套的数据结构
  formName?: string;
  children?: ReactNode;
}

export interface SFormItems extends ItemsProps {
  colProps?: ColProps;
  hidden?: boolean;
}

export interface SFormProps extends FormProps {
  rowProps?: RowProps;
  children?: ReactNode;
  items?: SFormItems[];
  columns?: number;
  // 是否为必填,提示文字
  required?: string | boolean;
  onFinish?: (e: any) => void;
  onReset?: (e: any) => void;
  // 只读模式
  readonly?: boolean;
  // 嵌套的数据结构
  formName?: string;
}

export type GroupItemsType = {
  container?: React.ComponentType<any>;
  columns?: number;
  title?: ReactNode;
  items?: SFormItems[];
  rowProps?: RowProps;
  // 嵌套的数据结构
  formName?: string;
};

// 分组表单
export interface SFormGroupProps extends FormProps {
  groupItems?: GroupItemsType[];
  onFinish?: (e: any) => void;
  onReset?: (e: any) => void;
  container?: React.ComponentType<any>;
  // 嵌套的数据结构
  formName?: string;
  children?: ReactNode;
  /** @description  只读模式 */
  readonly?: boolean;
}

export interface SearchProps extends SFormProps {
  // 默认是否展开
  defaultExpand?: boolean;
  showExpand?: boolean;
  // 展开收起的回调函数
  onExpand?: (expand: boolean) => void;
  // 展开的行数
  expandLine?: number;
  actionNode?: ReactNode;
  /** @description 自定义组件容器 */
  container?: React.ComponentType<any>;
  isCard?: boolean;
}
