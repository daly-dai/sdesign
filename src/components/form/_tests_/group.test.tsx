import { render } from '@testing-library/react';
import { Form, FormInstance } from 'antd';
import React, { useEffect } from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { describe, expect, it } from 'vitest';

import SForm from '../index';
import { GroupItemsType } from '../types';

import { waitFakeTimer } from '@daly/sdesign/tests/utils';

const dataSource = {
  userName: '张三',
  userPhone: '123456789',
  userMsg11: '张三',
  address: '北京市朝阳区',
  currentAddress: '北京市朝阳区',
};

const groupItems: GroupItemsType[] = [
  {
    title: '基础信息',
    columns: 2,
    items: [
      {
        type: 'input',
        label: '用户名称',
        name: 'userName',
        required: true,
      },
      {
        type: 'input',
        label: '用户电话',
        name: 'userPhone',
        required: '请输入用户电话',
      },
      {
        type: 'input',
        label: '用户信息',
        name: 'userMsg11',
        required: '请输入用户信息',
      },
    ],
  },
  {
    title: '详细信息',
    items: [
      {
        type: 'textarea',
        label: '户籍地址',
        name: 'address',
      },
      {
        type: 'textarea',
        label: '现居地址',
        name: 'currentAddress',
      },
    ],
  },
];

describe('SForm.Group', () => {
  /**
   * 支持groupItems,
   * 支持formName，
   * 支持dynamicContainer,
   * 支持children
   */
  it('should render correctly', () => {
    const { container } = render(<SForm.Group></SForm.Group>);

    expect(container).toMatchSnapshot();
  });

  it('should render correctly with groupItems', () => {
    const { container, getByText } = render(
      <Router>
        <SForm.Group groupItems={groupItems}></SForm.Group>,
      </Router>,
    );
    expect(container).toMatchSnapshot();
    const groupTitle = getByText('基础信息');
    expect(groupTitle).toBeInTheDocument();
  });

  it('should support formName', () => {
    let formInstance: FormInstance<any> | null = null;

    const Group = () => {
      const [form] = Form.useForm();

      useEffect(() => {
        formInstance = form;
      }, []);

      return (
        <SForm.Group
          formName="baseInfo"
          groupItems={groupItems}
          form={form}
        ></SForm.Group>
      );
    };

    render(
      <Router>
        <Group></Group>,
      </Router>,
    );

    waitFakeTimer(1000);
    // @ts-ignore
    formInstance?.setFieldsValue({ baseInfo: dataSource });

    waitFakeTimer(1000);

    // @ts-ignore
    const data = formInstance?.getFieldsValue();

    expect(data).toEqual({ baseInfo: dataSource });
  });

  it('should support  children', async () => {
    const { container } = render(
      <Router>
        <SForm.Group>submit</SForm.Group>
      </Router>,
    );

    waitFakeTimer(1000);

    expect(container).toMatchSnapshot();
  });
});
