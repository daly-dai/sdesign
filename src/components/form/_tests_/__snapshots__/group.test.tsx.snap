// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SForm.Group > should render correctly 1`] = `
<div>
  <form
    class="ant-form ant-form-vertical css-dev-only-do-not-override-3rel02"
  />
</div>
`;

exports[`SForm.Group > should render correctly with groupItems 1`] = `
<div>
  <form
    class="ant-form ant-form-vertical css-dev-only-do-not-override-3rel02"
  >
    <div
      class="ant-card ant-card-bordered css-dev-only-do-not-override-3rel02"
      style="margin-bottom: 16px; padding: 16px 24px; border-radius: 8px;"
    >
      <div
        class="ant-card-body"
        style="padding: 0px;"
      >
        <div
          class="acss-16ye7hh"
          style="margin-bottom: 16px;"
        >
          <div
            class="acss-70qvj9"
          >
            <span
              class="anticon acss-1y0qfxu"
              role="img"
            >
              <svg
                aria-hidden="true"
                class=""
                fill="currentColor"
                focusable="false"
                height="1em"
                version="1.1"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
                xmlns:xlink="http://www.w3.org/1999/xlink"
              >
                <g
                  fill="none"
                  fill-rule="evenodd"
                  id="页面-1"
                  stroke="none"
                  stroke-width="1"
                >
                  <g
                    id="组件-组件编目-基本信息"
                    transform="translate(-284.000000, -335.000000)"
                  >
                    <g
                      id="编组-4备份"
                      transform="translate(284.000000, 331.000000)"
                    >
                      <g
                        id="网络"
                        transform="translate(0.000000, 4.000000)"
                      >
                        <rect
                          fill="#000000"
                          fill-rule="nonzero"
                          height="16"
                          id="矩形"
                          opacity="0"
                          width="16"
                          x="0"
                          y="0"
                        />
                        <rect
                          fill="#2D7AF7"
                          height="16"
                          id="矩形"
                          width="2"
                          x="4"
                          y="0"
                        />
                        <rect
                          fill="#2D7AF7"
                          height="8"
                          id="矩形备份-3"
                          width="2"
                          x="9"
                          y="4"
                        />
                      </g>
                    </g>
                  </g>
                </g>
              </svg>
            </span>
            <div
              class="acss-13zb43w"
              style="font-size: 16px;"
            >
              基础信息
            </div>
          </div>
          <div
            class="acss-k008qs"
          />
        </div>
        <div
          class="ant-row css-dev-only-do-not-override-3rel02"
          style="margin-left: -12px; margin-right: -12px; row-gap: 16px;"
        >
          <div
            class="ant-col ant-col-12 css-dev-only-do-not-override-3rel02"
            style="padding-left: 12px; padding-right: 12px;"
          >
            <div
              class="ant-form-item css-dev-only-do-not-override-3rel02"
            >
              <div
                class="ant-row ant-form-item-row css-dev-only-do-not-override-3rel02"
              >
                <div
                  class="ant-col ant-form-item-label css-dev-only-do-not-override-3rel02"
                >
                  <label
                    class="ant-form-item-required ant-form-item-no-colon"
                    for="userName"
                    title="用户名称"
                  >
                    用户名称
                  </label>
                </div>
                <div
                  class="ant-col ant-form-item-control css-dev-only-do-not-override-3rel02"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <span
                        class="ant-input-affix-wrapper css-dev-only-do-not-override-3rel02 ant-input-outlined"
                      >
                        <input
                          aria-required="true"
                          class="ant-input css-dev-only-do-not-override-3rel02"
                          id="userName"
                          placeholder="请输入"
                          type="text"
                          value=""
                        />
                        <span
                          class="ant-input-suffix"
                        >
                          <span
                            class="ant-input-clear-icon ant-input-clear-icon-hidden"
                            role="button"
                            tabindex="-1"
                          >
                            <span
                              aria-label="close-circle"
                              class="anticon anticon-close-circle"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="close-circle"
                                fill="currentColor"
                                fill-rule="evenodd"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M512 64c247.4 0 448 200.6 448 448S759.4 960 512 960 64 759.4 64 512 264.6 64 512 64zm127.98 274.82h-.04l-.08.06L512 466.75 384.14 338.88c-.04-.05-.06-.06-.08-.06a.12.12 0 00-.07 0c-.03 0-.05.01-.09.05l-45.02 45.02a.2.2 0 00-.05.09.12.12 0 000 .07v.02a.27.27 0 00.06.06L466.75 512 338.88 639.86c-.05.04-.06.06-.06.08a.12.12 0 000 .07c0 .03.01.05.05.09l45.02 45.02a.2.2 0 00.09.05.12.12 0 00.07 0c.02 0 .04-.01.08-.05L512 557.25l127.86 127.87c.04.04.06.05.08.05a.12.12 0 00.07 0c.03 0 .05-.01.09-.05l45.02-45.02a.2.2 0 00.05-.09.12.12 0 000-.07v-.02a.27.27 0 00-.05-.06L557.25 512l127.87-127.86c.04-.04.05-.06.05-.08a.12.12 0 000-.07c0-.03-.01-.05-.05-.09l-45.02-45.02a.2.2 0 00-.09-.05.12.12 0 00-.07 0z"
                                />
                              </svg>
                            </span>
                          </span>
                        </span>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-col ant-col-12 css-dev-only-do-not-override-3rel02"
            style="padding-left: 12px; padding-right: 12px;"
          >
            <div
              class="ant-form-item css-dev-only-do-not-override-3rel02"
            >
              <div
                class="ant-row ant-form-item-row css-dev-only-do-not-override-3rel02"
              >
                <div
                  class="ant-col ant-form-item-label css-dev-only-do-not-override-3rel02"
                >
                  <label
                    class="ant-form-item-required ant-form-item-no-colon"
                    for="userPhone"
                    title="用户电话"
                  >
                    用户电话
                  </label>
                </div>
                <div
                  class="ant-col ant-form-item-control css-dev-only-do-not-override-3rel02"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <span
                        class="ant-input-affix-wrapper css-dev-only-do-not-override-3rel02 ant-input-outlined"
                      >
                        <input
                          aria-required="true"
                          class="ant-input css-dev-only-do-not-override-3rel02"
                          id="userPhone"
                          placeholder="请输入"
                          type="text"
                          value=""
                        />
                        <span
                          class="ant-input-suffix"
                        >
                          <span
                            class="ant-input-clear-icon ant-input-clear-icon-hidden"
                            role="button"
                            tabindex="-1"
                          >
                            <span
                              aria-label="close-circle"
                              class="anticon anticon-close-circle"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="close-circle"
                                fill="currentColor"
                                fill-rule="evenodd"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M512 64c247.4 0 448 200.6 448 448S759.4 960 512 960 64 759.4 64 512 264.6 64 512 64zm127.98 274.82h-.04l-.08.06L512 466.75 384.14 338.88c-.04-.05-.06-.06-.08-.06a.12.12 0 00-.07 0c-.03 0-.05.01-.09.05l-45.02 45.02a.2.2 0 00-.05.09.12.12 0 000 .07v.02a.27.27 0 00.06.06L466.75 512 338.88 639.86c-.05.04-.06.06-.06.08a.12.12 0 000 .07c0 .03.01.05.05.09l45.02 45.02a.2.2 0 00.09.05.12.12 0 00.07 0c.02 0 .04-.01.08-.05L512 557.25l127.86 127.87c.04.04.06.05.08.05a.12.12 0 00.07 0c.03 0 .05-.01.09-.05l45.02-45.02a.2.2 0 00.05-.09.12.12 0 000-.07v-.02a.27.27 0 00-.05-.06L557.25 512l127.87-127.86c.04-.04.05-.06.05-.08a.12.12 0 000-.07c0-.03-.01-.05-.05-.09l-45.02-45.02a.2.2 0 00-.09-.05.12.12 0 00-.07 0z"
                                />
                              </svg>
                            </span>
                          </span>
                        </span>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-col ant-col-12 css-dev-only-do-not-override-3rel02"
            style="padding-left: 12px; padding-right: 12px;"
          >
            <div
              class="ant-form-item css-dev-only-do-not-override-3rel02"
            >
              <div
                class="ant-row ant-form-item-row css-dev-only-do-not-override-3rel02"
              >
                <div
                  class="ant-col ant-form-item-label css-dev-only-do-not-override-3rel02"
                >
                  <label
                    class="ant-form-item-required ant-form-item-no-colon"
                    for="userMsg11"
                    title="用户信息"
                  >
                    用户信息
                  </label>
                </div>
                <div
                  class="ant-col ant-form-item-control css-dev-only-do-not-override-3rel02"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <span
                        class="ant-input-affix-wrapper css-dev-only-do-not-override-3rel02 ant-input-outlined"
                      >
                        <input
                          aria-required="true"
                          class="ant-input css-dev-only-do-not-override-3rel02"
                          id="userMsg11"
                          placeholder="请输入"
                          type="text"
                          value=""
                        />
                        <span
                          class="ant-input-suffix"
                        >
                          <span
                            class="ant-input-clear-icon ant-input-clear-icon-hidden"
                            role="button"
                            tabindex="-1"
                          >
                            <span
                              aria-label="close-circle"
                              class="anticon anticon-close-circle"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="close-circle"
                                fill="currentColor"
                                fill-rule="evenodd"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M512 64c247.4 0 448 200.6 448 448S759.4 960 512 960 64 759.4 64 512 264.6 64 512 64zm127.98 274.82h-.04l-.08.06L512 466.75 384.14 338.88c-.04-.05-.06-.06-.08-.06a.12.12 0 00-.07 0c-.03 0-.05.01-.09.05l-45.02 45.02a.2.2 0 00-.05.09.12.12 0 000 .07v.02a.27.27 0 00.06.06L466.75 512 338.88 639.86c-.05.04-.06.06-.06.08a.12.12 0 000 .07c0 .03.01.05.05.09l45.02 45.02a.2.2 0 00.09.05.12.12 0 00.07 0c.02 0 .04-.01.08-.05L512 557.25l127.86 127.87c.04.04.06.05.08.05a.12.12 0 00.07 0c.03 0 .05-.01.09-.05l45.02-45.02a.2.2 0 00.05-.09.12.12 0 000-.07v-.02a.27.27 0 00-.05-.06L557.25 512l127.87-127.86c.04-.04.05-.06.05-.08a.12.12 0 000-.07c0-.03-.01-.05-.05-.09l-45.02-45.02a.2.2 0 00-.09-.05.12.12 0 00-.07 0z"
                                />
                              </svg>
                            </span>
                          </span>
                        </span>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-card ant-card-bordered css-dev-only-do-not-override-3rel02"
      style="margin-bottom: 16px; padding: 16px 24px; border-radius: 8px;"
    >
      <div
        class="ant-card-body"
        style="padding: 0px;"
      >
        <div
          class="acss-16ye7hh"
          style="margin-bottom: 16px;"
        >
          <div
            class="acss-70qvj9"
          >
            <span
              class="anticon acss-1y0qfxu"
              role="img"
            >
              <svg
                aria-hidden="true"
                class=""
                fill="currentColor"
                focusable="false"
                height="1em"
                version="1.1"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
                xmlns:xlink="http://www.w3.org/1999/xlink"
              >
                <g
                  fill="none"
                  fill-rule="evenodd"
                  id="页面-1"
                  stroke="none"
                  stroke-width="1"
                >
                  <g
                    id="组件-组件编目-基本信息"
                    transform="translate(-284.000000, -335.000000)"
                  >
                    <g
                      id="编组-4备份"
                      transform="translate(284.000000, 331.000000)"
                    >
                      <g
                        id="网络"
                        transform="translate(0.000000, 4.000000)"
                      >
                        <rect
                          fill="#000000"
                          fill-rule="nonzero"
                          height="16"
                          id="矩形"
                          opacity="0"
                          width="16"
                          x="0"
                          y="0"
                        />
                        <rect
                          fill="#2D7AF7"
                          height="16"
                          id="矩形"
                          width="2"
                          x="4"
                          y="0"
                        />
                        <rect
                          fill="#2D7AF7"
                          height="8"
                          id="矩形备份-3"
                          width="2"
                          x="9"
                          y="4"
                        />
                      </g>
                    </g>
                  </g>
                </g>
              </svg>
            </span>
            <div
              class="acss-13zb43w"
              style="font-size: 16px;"
            >
              详细信息
            </div>
          </div>
          <div
            class="acss-k008qs"
          />
        </div>
        <div
          class="ant-row css-dev-only-do-not-override-3rel02"
          style="margin-left: -12px; margin-right: -12px; row-gap: 16px;"
        >
          <div
            class="ant-col ant-col-24 css-dev-only-do-not-override-3rel02"
            style="padding-left: 12px; padding-right: 12px;"
          >
            <div
              class="ant-form-item css-dev-only-do-not-override-3rel02"
            >
              <div
                class="ant-row ant-form-item-row css-dev-only-do-not-override-3rel02"
              >
                <div
                  class="ant-col ant-form-item-label css-dev-only-do-not-override-3rel02"
                >
                  <label
                    class="ant-form-item-no-colon"
                    for="address"
                    title="户籍地址"
                  >
                    户籍地址
                  </label>
                </div>
                <div
                  class="ant-col ant-form-item-control css-dev-only-do-not-override-3rel02"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <span
                        class="ant-input-affix-wrapper ant-input-textarea-affix-wrapper css-dev-only-do-not-override-3rel02 ant-input-textarea-allow-clear ant-input-outlined"
                      >
                        <textarea
                          class="ant-input css-dev-only-do-not-override-3rel02"
                          id="address"
                          placeholder="请输入"
                        />
                        <span
                          class="ant-input-suffix"
                        >
                          <span
                            class="ant-input-clear-icon ant-input-clear-icon-hidden"
                            role="button"
                            tabindex="-1"
                          >
                            <span
                              aria-label="close-circle"
                              class="anticon anticon-close-circle"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="close-circle"
                                fill="currentColor"
                                fill-rule="evenodd"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M512 64c247.4 0 448 200.6 448 448S759.4 960 512 960 64 759.4 64 512 264.6 64 512 64zm127.98 274.82h-.04l-.08.06L512 466.75 384.14 338.88c-.04-.05-.06-.06-.08-.06a.12.12 0 00-.07 0c-.03 0-.05.01-.09.05l-45.02 45.02a.2.2 0 00-.05.09.12.12 0 000 .07v.02a.27.27 0 00.06.06L466.75 512 338.88 639.86c-.05.04-.06.06-.06.08a.12.12 0 000 .07c0 .03.01.05.05.09l45.02 45.02a.2.2 0 00.09.05.12.12 0 00.07 0c.02 0 .04-.01.08-.05L512 557.25l127.86 127.87c.04.04.06.05.08.05a.12.12 0 00.07 0c.03 0 .05-.01.09-.05l45.02-45.02a.2.2 0 00.05-.09.12.12 0 000-.07v-.02a.27.27 0 00-.05-.06L557.25 512l127.87-127.86c.04-.04.05-.06.05-.08a.12.12 0 000-.07c0-.03-.01-.05-.05-.09l-45.02-45.02a.2.2 0 00-.09-.05.12.12 0 00-.07 0z"
                                />
                              </svg>
                            </span>
                          </span>
                        </span>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-col ant-col-24 css-dev-only-do-not-override-3rel02"
            style="padding-left: 12px; padding-right: 12px;"
          >
            <div
              class="ant-form-item css-dev-only-do-not-override-3rel02"
            >
              <div
                class="ant-row ant-form-item-row css-dev-only-do-not-override-3rel02"
              >
                <div
                  class="ant-col ant-form-item-label css-dev-only-do-not-override-3rel02"
                >
                  <label
                    class="ant-form-item-no-colon"
                    for="currentAddress"
                    title="现居地址"
                  >
                    现居地址
                  </label>
                </div>
                <div
                  class="ant-col ant-form-item-control css-dev-only-do-not-override-3rel02"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <span
                        class="ant-input-affix-wrapper ant-input-textarea-affix-wrapper css-dev-only-do-not-override-3rel02 ant-input-textarea-allow-clear ant-input-outlined"
                      >
                        <textarea
                          class="ant-input css-dev-only-do-not-override-3rel02"
                          id="currentAddress"
                          placeholder="请输入"
                        />
                        <span
                          class="ant-input-suffix"
                        >
                          <span
                            class="ant-input-clear-icon ant-input-clear-icon-hidden"
                            role="button"
                            tabindex="-1"
                          >
                            <span
                              aria-label="close-circle"
                              class="anticon anticon-close-circle"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="close-circle"
                                fill="currentColor"
                                fill-rule="evenodd"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M512 64c247.4 0 448 200.6 448 448S759.4 960 512 960 64 759.4 64 512 264.6 64 512 64zm127.98 274.82h-.04l-.08.06L512 466.75 384.14 338.88c-.04-.05-.06-.06-.08-.06a.12.12 0 00-.07 0c-.03 0-.05.01-.09.05l-45.02 45.02a.2.2 0 00-.05.09.12.12 0 000 .07v.02a.27.27 0 00.06.06L466.75 512 338.88 639.86c-.05.04-.06.06-.06.08a.12.12 0 000 .07c0 .03.01.05.05.09l45.02 45.02a.2.2 0 00.09.05.12.12 0 00.07 0c.02 0 .04-.01.08-.05L512 557.25l127.86 127.87c.04.04.06.05.08.05a.12.12 0 00.07 0c.03 0 .05-.01.09-.05l45.02-45.02a.2.2 0 00.05-.09.12.12 0 000-.07v-.02a.27.27 0 00-.05-.06L557.25 512l127.87-127.86c.04-.04.05-.06.05-.08a.12.12 0 000-.07c0-.03-.01-.05-.05-.09l-45.02-45.02a.2.2 0 00-.09-.05.12.12 0 00-.07 0z"
                                />
                              </svg>
                            </span>
                          </span>
                        </span>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
  ,
</div>
`;

exports[`SForm.Group > should support  children 1`] = `<div />`;

exports[`SForm.Group > should support dynamicContainer 1`] = `<div />`;
